service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/CodeableConceptCdMapper.java

```diff
@@ -77,12 +77,16 @@ public String mapCodeableConceptToCdForTransformedActualProblemHeader(CodeableCo
                 return Optional.of(codeableConcept.getText());
             } else {
                 var extension = retrieveDisplayExtension(coding.get());
-                return extension.map(value -> value
-                    .getExtension()
-                    .stream()
+                Optional<String> originalText = extension.stream()
                     .filter(extension1 -> extension1.getUrl().equals(DESCRIPTION_DISPLAY))
```

Comment:
Just a small one, might not have even been your change. 

rename extension1 to extension, if needs be then renamed the current extension variable to displayExtension